File: Leetcode/All Elements in Two Binary Search Trees/All Elements in Two Binary Search Trees.java

// Problem Statement:
//  All Elements in Two Binary Search Trees
//     Given two binary search trees root1 and root2.
//     Return a list containing all the integers from both trees sorted in ascending order.

 
class Solution {
    // Create an empty arrayList a: this will contain the inorder traversed values
    List<Integer> a = new ArrayList();  
    public List<Integer> getAllElements(TreeNode root1, TreeNode root2) {
        // Find the inorder traversal for each tree: This will sort the values for each node
        inorder(root1);
        inorder(root2);
        List<Integer> ans = new ArrayList(a);
        // sort the list
        Collections.sort(ans);
        return ans;
    }
    public void inorder(TreeNode root){
        // Base condition
        if(root == null)
            return;
        // for inorder traversal: left -> root ->  right
        inorder(root.left);
        a.add(root.val);
        inorder(root.right);
    }
}

File: Leetcode/All Elements in Two Binary Search Trees/All Elements in Two Binary Search Trees.py
class Solution:
    def getAllElements(self, root1: TreeNode, root2: TreeNode) -> List[int]:        # Getting elements of both the trees
        def dfs(root):                                                              # Using recursion to find root
            if not root:
                return
            dfs(root.left)
            l.append(root.val)                                                     # Storing values of the tree in the list
            dfs(root.right)
        l=[]                                                                       # Creating empty list to store values
        dfs(root1)
        dfs(root2)
        return sorted(l)                                                           # Giving result after sorting the list
